File: spoj/marbles.cpp
/*******************************************************************************
 * Author       : Kashyap R Puranik
 * email        : kashyap.r.puranik (at) gmail (dot) com
 * copyright    : 2008 - 2009
 * date         : 23 - 07 -09
 * 
 * file name    : ZofN.cpp 
 * version      : 1.0.1
 ******************************************************************************/
 
/*******************************************************************************
 * some pointers
 *
*******************************************************************************/

#include <stdio.h>
#include <stdlib.h>

long long int C( long long int x, long long int y) {
	if ( y == 0 )
		return 1;
	if ( x == y )
		return 1;
	else
		return (C (x - 1, y) + C (x - 1, y - 1));
}
int main()
{
	long int t = 0, T;			// This is for the test cases
	scanf("%ld",&T);
	while ( t++ < T )				// go through all test cases
	{
		long long int N, R;
		// get the two numbers.
		scanf("%lld",&N);
		scanf("%lld",&R);
		
		printf("%lld\n",C(N-1, N-R));
		
	}
	return 0;
}

File: spoj/marbles.java
/*******************************************************************************
 * Author       : Kashyap R Puranik
 * email        : kashyap.r.puranik (at) gmail (dot) com
 * copyright    : 2008 - 2009
 * date         : 23 - 08 -09
 * 
 * file name    : .cpp 
 * version      : 1.0.1
 ******************************************************************************/
import java.util.Scanner;
import java.io.*;
import java.math.BigInteger;

public class Main {
	

    public static BigInteger comb (long n, long r) {
        BigInteger result = new BigInteger("1");
        for (long i = n; i > n-r; i--) {
            result = result.multiply(BigInteger.valueOf(i));
        }
        for ( long i = r; i > 0 ; i--) {
        	result = result.divide(BigInteger.valueOf(i));
        }
        return result;
    }
	
	// Control flow starts here
	public static void main(String[] args) {

		int num, i;
		Scanner in = new Scanner(System.in);
		
		num=in.nextInt();						// Get the number of test cases
		for ( i = 0 ; i < num ; i ++ )
		{
			long x, y;
			x = in.nextInt();
			y = in.nextInt();

			System.out.println(comb(x-1,y-1));
		}
		in.close();
	}
}

